schism/video.c

```diff
@@ -250,6 +250,10 @@ void video_startup(void)
 
 	video_setup(cfg_video_interpolation);
 
+#ifndef SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR
+/* older SDL2 versions don't define this, don't fail the build for it */
+#define SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR "SDL_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR"
+#endif
 	SDL_SetHint(SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR, "0");
```

Comment:
```suggestion
#ifdef SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR
	SDL_SetHint(SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR, "0");
#endif
```

Reply:
The intention isn't to discard the hint just because the build environment is missing the define... (assuming the hint is desirable)

Keep in mind via dynamic linking the actual runtime sdl2 version can be newer than the build version.

Reply:
> Keep in mind via dynamic linking the actual runtime sdl2 version can be newer than the build version.
Ah, I had completely forgotten about that. In that case, it's probably better to just do what you've done instead of doing some `SDL_GetVersion()` shenanigans. Disregard this.

Reply:
Plus since it's just a #define, the string it maps to forms the actual ABI contract SDL2 is committing to.  So those hint defines are kind of just provided as a convenience, and we're in the clear defining it ourselves provided the string it maps to is the same.